nifpga/bitfile.py

```diff
@@ -28,20 +32,25 @@ def __init__(self, filepath, parse_contents=False):
         self._base_address_on_device = int(nifpga.find("BaseAddressOnDevice").text)
         self._registers = {}
         for reg_xml in tree.find("VI").find("RegisterList"):
-            reg = self.create_register(reg_xml)
-            if reg.datatype is not None:
+            try:
+                reg = Register(reg_xml)
                 assert reg.name not in self._registers, \
                     "One or more registers have the same name '%s', this is not supported" % reg.name
-                # We don't yet support FXP Arrays
-                if not (reg.is_array() and reg.datatype is DataType.Fxp):
-                    self._registers[reg.name] = reg
+                self._registers[reg.name] = reg
+            except UnsupportedTypeError as e:
+                warn("Skipping Register: %s, %s" % (reg.name, str(e)))
+            except ClusterMustContainUniqueNames as e:
```

Comment:
I think the cluster fields are also numbered, and those numbers are guaranteed unique. Those numbers don't seem to appear in the XML datatype, but I suppose a future version of the .lvbitx could include the field numbers. I suppose the XML might be printed in an order that corresponds to those numbers.